compiler/rustc_hir_typeck/src/fn_ctxt/suggestions.rs

```diff
@@ -1512,9 +1514,118 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         found_ty: Ty<'tcx>,
         expr: &hir::Expr<'_>,
     ) {
+        // When `expr` is `x` in something like `let x = foo.clone(); x`, need to recurse up to
+        // get `foo` and `clone`.
+        match expr.kind {
+            hir::ExprKind::Path(hir::QPath::Resolved(
+                None,
+                hir::Path { segments: [_], res: crate::Res::Local(binding), .. },
+            )) => {
+                let Some(hir::Node::Pat(hir::Pat { hir_id, .. })) = self.tcx.hir().find(*binding)
+                else {
```

Comment:
Can you extract this to a separate function that returns the relevant expression instead?

Reply:
just to be clear I'd like you to extract this `match` statement into a function that could possibly return an expression corresponding to the method call, or return the original expression

Reply:
Covered in latest commit